src/fsharp/service/IncrementalBuild.fs

```diff
@@ -1022,16 +1023,91 @@ module IncrementalBuilderEventTesting =
 
 module Tc = FSharp.Compiler.TypeChecker
 
+[<AutoOpen>]
+module IncrementalBuildSyntaxTree =
```

Comment:
I know this is internal, but I think there needs to be some comments explaining why we need our own abstraction over a syntax tree for the incremental builder. If I were to need to change something in here, I'd have nothing to look at for context on why this exists.

Reply:
We need some kind of object type that wraps the information to lazily parse the source text to get a `ParsedInput`(which is the root of the syntax tree). The name `SyntaxTree` feels correct here.